Document HTTP Header manipulation options added in #10613 (#11037)
hashicorp/consul: website/content/docs/connect/config-entries/service-router.mdx

      description:
        'A list of HTTP response status codes that are eligible for retry.',
    },
    {
      yaml: false,
      name: 'RequestHeaders',
      type: 'HTTPHeaderModifiers: <optional>',

>       description: `A set of [HTTP-specific header modification rules](/docs/connect/config-entries/service-router#httpheadermodifiers)

Reviewer: ```suggestion
      description: `A set of [HTTP-specific header modification rules](#httpheadermodifiers)
```
minor suggestion: since this references an object on the same page, you can remove the URL path and just leave the fragment.

Author: I might leave this on the basis that this block is copy pasted between three config entries right now. If we remove the URL on this page it makes it harder to copy paste between them for future updates?

      that will be applied to requests routed to this service.
      This cannot be used with a \`tcp\` listener.`,
    },
    {
      yaml: false,
      name: 'ResponseHeaders',
      type: 'HTTPHeaderModifiers: <optional>',
      description: `A set of [HTTP-specific header modification rules](/docs/connect/config-entries/service-router#httpheadermodifiers)
      that will be applied to responses from this service.
      This cannot be used with a \`tcp\` listener.`,
    },
  ]}
/>

### `HTTPHeaderModifiers`

<ConfigEntryReference
  topLevel={false}
  yaml={false}
  keys={[
    {
      hcl: false,
      name: 'Unsupported',
      type: '',
      description: `HTTP Header modification is not yet supported in our Kubernetes CRDs.`,
    },
    {
      yaml: false,
      name: 'Add',
      type: 'map<string|string>: optional',

>       description: `The set of header value keyed by header name to
>       add. If a header with the same (case-insensitive) name already

Reviewer: ```suggestion
      description: `The set of header values to add, keyed by header name.
      If a header with the same (case-insensitive) name already
```

>       description: `The set of header values to add, keyed by header name.
>       If a header with the same (case-insensitive) name already

Reviewer 2: ```suggestion
      description: `The set of key/value pairs that specify header values to add. Use header names as keys. Header names are case-sensitive.
      If a header with the same name already
```
Just thinking about non-native English speakers--not sure if 'keyed by header name' would make sense. Also, case-sensitivity seems important enough to call out as its own statement.

      exists, the value set here will be appended and both will be present.
      If Envoy is used as the proxy, the value may contain
      [variable placeholders](https://www.envoyproxy.io/docs/envoy/latest/configuration/http/http_conn_man/headers#custom-request-response-headers) for example
      \`%DOWNSTREAM_REMOTE_ADDRESS%\` to interpolate dynamic request
      metadata into the value added.`,
    },
    {
      yaml: false,
      name: 'Set',
      type: 'map<string|string>: optional',

>       description: `The set of header value keyed by header name to

Reviewer 2: Same comment as line 613.

      add. If one or more header values with the same (case-insensitive) name already exist,
      the value set here will replace them all.
      If Envoy is used as the proxy, the value may contain
      [variable placeholders](https://www.envoyproxy.io/docs/envoy/latest/configuration/http/http_conn_man/headers#custom-request-response-headers) for example
      \`%DOWNSTREAM_REMOTE_ADDRESS%\` to interpolate dynamic request
      metadata into the value added.`,
    },
    {
      yaml: false,
      name: 'Remove',
      type: 'array<string>: optional',
      description: `The set of header names to remove. Only headers

>       whose names are an <i>case-insensitive</i> exact match will be removed`,

Reviewer: ```suggestion
      whose names are a <i>case-insensitive</i> exact match will be removed`,
```
